pkg/download/stasher/basic.go

```diff
@@ -0,0 +1,56 @@
+package stasher
+
+import (
+	"context"
+	"time"
+
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/storage"
+)
+
+// Stasher has the job of taking a module
+// from its upstream Download Protcool and
+// stashing to its Storage Backend. Both
+// interfaces must be defined on the type previously.
+type Stasher interface {
+	Stash(mod, ver string) error
+}
+
+// DPVersionFn is a version func of download protocol
+type DPVersionFn func(context.Context, string, string) (*storage.Version, error)
```

Comment:
this is basically what the module.Fetcher should *really* be. The module.Fetcher is a two step process for no reason. 

We should just have 

```golang
type Fetcher interface {
  Fetch(context.Context, string, string) (*storage.Version, error)
}
```

Reply:
yes, we can even omit Version

Reply:
i believe you already did that in your updated version, i worked with the one before the update